Assets/Scripts/CreatureStats.cs

```diff
@@ -133,19 +140,70 @@ private void DisplayPopup(string text)
         textLabelText = text;
     }
 
+    // Returns true if floats are within 15.0f of each other.
+    // compare Mathf.Approximately.
+    private bool VeryApproximately(float f1, float f2)
+    {
+        if (Mathf.Abs(f1 - f2) < 15.0f) return true;
+        return false;
+    }
+
+    // Returns true if the target is being attacked from the side or rear.
+    // Side/read attacks double the chance of a critical hit.
+    public bool IsFlanking(CreatureStats target)
+    {
+        float rotation1 = target.transform.eulerAngles.y;
+        float rotation2 = transform.eulerAngles.y;
+        bool flanking = !VeryApproximately(Mathf.Abs(rotation1 - rotation2), 180.0f);
+        return flanking;
+    }
+
+    // Returns true if the target is being attacked from the rear.
+    // Rear attacks do a bonus 0-10 damage plus one half of agility.
+    public bool IsBehind(CreatureStats target)
+    {
+        float rotation1 = target.transform.eulerAngles.y;
+        float rotation2 = transform.eulerAngles.y;
+        bool behind = VeryApproximately(Mathf.Abs(rotation1 - rotation2), 0.0f) || VeryApproximately(Mathf.Abs(rotation1 - rotation2), 360.0f);
+        return behind;
+    }
+
+    public bool IsCrit(CreatureStats target)
+    {
+        int chance = GetEffectiveIntelligence() / 10;
+        if (IsFlanking(target))
+        {
+            chance *= 2;
+        }
+        return PercentRoll(chance);
+    }
+
     public void PerformAttack(CreatureStats target)
     {
-        if (!percentRoll(hitChance())) {
+        if (!PercentRoll(HitChance())) {
             DisplayPopup("Miss!");
             return;
         }
-        if (percentRoll(target.dodgeChance()))
+        if (PercentRoll(target.DodgeChance()))
         {
             target.DisplayPopup("Dodge!");
             return;
         }
-        int dam = damageInflicted();
-        target.DisplayPopup(dam + " damage inflicted!");
+        int dam = DamageInflicted();
+        if (IsBehind(target))
+        {
+            DisplayPopup("Rear attack!");
+            dam += BonusRearDamage();
+        }
+        if (IsCrit(target))
+        {
+            target.DisplayPopup("CRITICAL HIT! " + dam + " damage inflicted!");
```

Comment:
Should this go after you add another 50% to the damage?

Reply:
Yeah - in this case, stacking multipliers is the intended behavior. Though we can always revise it later if need-be.